Clarify BQ behaviour for incremental merge updates. (#3430)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/bigquery-configs.md

when not matched then insert ...
```

The 'merge' approach automatically updates new data in the destination incremental table but requires scanning all source tables referenced in the model SQL, as well as destination tables. This can be slow and expensive for large data volumes. [Partitioning and clustering](#using-table-partitioning-and-clustering) techniques mentioned earlier can help mitigate these issues.

> of scanning both source and destination tables can be mitigated through the use of partitioning

Reviewer: ```suggestion
```

when not matched then insert ...
```

> destination incremental table. The drawback of this approach is that BigQuery must scan all

Reviewer: ```suggestion
```

when not matched then insert ...
```

The 'merge' approach automatically updates new data in the destination incremental table but requires scanning all source tables referenced in the model SQL, as well as destination tables. This can be slow and expensive for large data volumes. [Partitioning and clustering](#using-table-partitioning-and-clustering) techniques mentioned earlier can help mitigate these issues.

> transforming very large amounts of data.

Reviewer: ```suggestion
```

when not matched then insert ...
```

The 'merge' approach automatically updates new data in the destination incremental table but requires scanning all source tables referenced in the model SQL, as well as destination tables. This can be slow and expensive for large data volumes. [Partitioning and clustering](#using-table-partitioning-and-clustering) techniques mentioned earlier can help mitigate these issues.

> and clustering as described above).  This can be slow and costly if the incremental model is

Reviewer: ```suggestion
```

> The `merge` approach has the benefit of automatically updating any late-arriving facts in the

Reviewer: ```suggestion
```

destination incremental table. The drawback of this approach is that BigQuery must scan all

> source tables referenced in the model SQL, as well as the the destination table (however the cost

Reviewer: ```suggestion
The 'merge' approach automatically updates new data in the destination incremental table but requires scanning all source tables referenced in the model SQL, as well as destination tables. This can be slow and expensive for large data volumes. [Partitioning and clustering](#using-table-partitioning-and-clustering) techniques mentioned earlier can help mitigate these issues.
```
